drivers/soundwire/intel.c

```diff
@@ -1005,9 +1005,13 @@ static int intel_trigger(struct snd_pcm_substream *substream, int cmd, struct sn
 {
 	struct sdw_cdns *cdns = snd_soc_dai_get_drvdata(dai);
 	struct sdw_intel *sdw = cdns_to_intel(cdns);
+	struct sdw_intel_link_res *res = sdw->link_res;
 	struct sdw_cdns_dma_data *dma;
 	int ret = 0;
 
+	if (res->ops && res->ops->trigger)
+		res->ops->trigger(substream, cmd, dai);
+
```

Comment:
it's not super-elegant, we already have an intel_free_stream() below to take care of the TRIGGER_SUSPEND case.

Reply:
free_stream ops  is pretty specific to free a stream and it needs some specific parameters, I'm not sure if that could be handled via this new callback also.